### PR TITLE
Code for Lucas-Kanade method for tracking

### LINKED ISSUE #1381: Implement classical object tracking algorithms like Lucas-Kande and KCF
## 🚀 Feature

Since we have started our tracking module, it would be nice to populate it with classical and CNN-based things

References:

- http://robots.stanford.edu/cs223b04/algo_tracking.pdf
- [Lukas-Kanade Tracker](https://www.ri.cmu.edu/pub_files/pub3/baker_simon_2002_3/baker_simon_2002_3.pdf)
- [KCF](https://cw.fel.cvut.cz/b202/courses/mpv/labs/4b_tracking/start)

### PR DESCRIPTION
**Pull request by AnimeshMaheshwari22**
#### Changes
Adding the code for Lucas-Kanade method for tracking

Fixes #1381 

#### Type of change
- [X] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: kornia/tracking/lucaskanadetracking.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import torch
+import torch.nn as nn
+
+from kornia.filters import filter2d
+
+
+def lucaskanadetracker(input: torch.Tensor) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (AnimeshMaheshwari22) [reply kornia/tracking/lucaskanadetracking.py:7]: Currently the code takes a stacked tensor representing 2 frames and calculates velocity vectors between them. I wanted to clarify this so had asked a question regarding this.

**Comment by ducha-aiki** [comment]
@AnimeshMaheshwari22 sorry, I have missed your question then. 
The stacking makes sense now, although I would propose a different solution:

1)Stack not in batch dimension, but similarly to what we have for scalespace and video: 5d tensor [B, Ch, time, H, W]. This way you can run in batches
2) use kornia.filters.spatial_gradient_3d, which does all the gradients you have implemented manually

This way you will have Lukas- Kanade optical flow the whole image. Useful thing on its own (worth separate PR), but not tracking, as we are still missing target. 
And target can come in form of keypoints, bounding box or even segmentation.

File: kornia/tracking/lucaskanadetracking.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (edgarriba) [reply kornia/tracking/lucaskanadetracking.py:7]: i agree here that to have a clean api we should input separately the previous and current frames. I would target something similar to the opencv implementation: https://docs.opencv.org/3.4/dc/d6b/group__video__track.html#ga473e4b886d0bcc6b65831eb88ed93323

```python
def optical_flow_lk(image_prev: torch.Tensor, image_next: torch.Tensor, ...) -> torch.Tensor:
   """Calculate an optical flow for an input pair of images.

     Args:
        image_prev: the image tensor in the previous time-frame with shape :math:`(B, C, H, W)`.
        image_next: the image tensor in the current time-frame with shape :math:`(B, C, H, W)`.

      Return:
        The optical flow image containing the motion vector [x y] per pixel with shape :math:`(B, H, W, 2)`.
   """
```
@ducha-aiki interface suggestions here ?


```

[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/tracking/lucaskanadetracking.py:7]: @edgarriba  I like your suggestion

**Comment by AnimeshMaheshwari22** [comment]
> @AnimeshMaheshwari22 sorry, I have missed your question then. The stacking makes sense now, although I would propose a different solution:
> 
> 1)Stack not in batch dimension, but similarly to what we have for scalespace and video: 5d tensor [B, Ch, time, H, W]. This way you can run in batches 2) use kornia.filters.spatial_gradient_3d, which does all the gradients you have implemented manually
> 
> This way you will have Lukas- Kanade optical flow the whole image. Useful thing on its own (worth separate PR), but not tracking, as we are still missing target. And target can come in form of keypoints, bounding box or even segmentation.

I misunderstood the problem statement and this approach would be benificial in optical flow, referring to issue #778 . So I will close this and open another PR for that issue, while I will also work on KLT tracker. 
Also I have a doubt. Using kornia.filters.spatial_gradient_3d would convolve with specific filters right? but for optical flow with the Lucas-Kanade method require calculating gradients with different masks right?
